File: src/test/java/se/johannesdahlgren/adventofcode2021/Day9Test.java
package se.johannesdahlgren.adventofcode2021;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day9Test {

  @Test
  void part1Example() {
    Day9 day9 = new Day9("day9example.txt");
    var riskLevel = day9.getRiskLevelsSum();
    assertThat(riskLevel).isEqualTo(15L);
  }

  @Test
  void part1() {
    Day9 day9 = new Day9("day9.txt");
    var riskLevel = day9.getRiskLevelsSum();
    assertThat(riskLevel).isEqualTo(468L);
  }

  @Test
  void part2Example() {
    Day9 day9 = new Day9("day9example.txt");
    var riskLevel = day9.getBasinSizes();
    assertThat(riskLevel).isEqualTo(1134L);
  }

  @Test
  void part2() {
    Day9 day9 = new Day9("day9.txt");
    var riskLevel = day9.getBasinSizes();
    assertThat(riskLevel).isEqualTo(1280496L);
  }

}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day9Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day9Test {
    @Test
    fun part1Example() {
        val day9 = Day9("day9Example")
        assertThat(day9.calcTailUniqueVisits(2)).isEqualTo(13)
    }

    @Test
    fun part1() {
        val day9 = Day9("day9")
        assertThat(day9.calcTailUniqueVisits(2)).isEqualTo(6337)
    }

    @Test
    fun part2Example() {
        val day9 = Day9("day9Example")
        assertThat(day9.calcTailUniqueVisits(10)).isEqualTo(1)
    }

    @Test
    fun part2Example2() {
        val day9 = Day9("day9Example2")
        assertThat(day9.calcTailUniqueVisits(10)).isEqualTo(36)
    }

    @Test
    fun part2() {
        val day9 = Day9("day9")
        assertThat(day9.calcTailUniqueVisits(10)).isEqualTo(2455)
    }
}